src/main/java/com/ft/methodecontentcollectionmapper/mapping/EomFileToContentCollectionMapper.java

```diff
@@ -0,0 +1,88 @@
+package com.ft.methodecontentcollectionmapper.mapping;
+
+import com.ft.methodecontentcollectionmapper.exception.TransformationException;
+import com.ft.methodecontentcollectionmapper.exception.UnsupportedTypeException;
+import com.ft.methodecontentcollectionmapper.model.ContentCollection;
+import com.ft.methodecontentcollectionmapper.model.ContentCollectionType;
+import com.ft.methodecontentcollectionmapper.model.EomFile;
+import com.ft.methodecontentcollectionmapper.model.EomLinkedObject;
+import com.ft.methodecontentcollectionmapper.model.Item;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+public class EomFileToContentCollectionMapper {
+
+  private static final String XPATH_CONTAINER_WEB_TYPE = "ObjectMetadata/FTcom/DIFTcomWebType";
+
+  private static final String STORY_PACKAGE_WEB_TYPE = "editorsChoice";
+  private static final String CONTENT_PACKAGE_WEB_TYPE = "content-package";
+
+  public ContentCollection mapPackage(EomFile eomFile, String transactionId, Date lastModified) {
+    try {
+      XPath xpath = XPathFactory.newInstance().newXPath();
+      Document attributesDocument = getDocumentBuilder()
+          .parse(new InputSource(new StringReader(eomFile.getAttributes())));
+      String webType = xpath.evaluate(XPATH_CONTAINER_WEB_TYPE, attributesDocument);
+      verifyWebType(webType, eomFile.getUuid());
+
+      ContentCollectionType type = ContentCollectionType.STORY_PACKAGE_TYPE;
```

Comment:
Yeah, it's a bit of duplicate checks, bad for maintenance when you want to add one more type. I've did refactored this and it's much better now. Thanks for the suggestion :smiley: